packages/progress-bar/src/ProgressBar/ProgressBar.tsx

```diff
@@ -179,12 +179,19 @@ export function ProgressBar(props: ProgressBarProps) {
           >
             <div
               data-lgid={lgIds.fill}
+              style={
+                {
+                  // to avoid rebuilding styles on every width change, use CSS variables
+                  ...(!isIndeterminate && {
+                    '--width': `${displayWidth}%`,
+                    '--width-animation-duration': `${widthAnimationDuration}ms`,
+                  }),
+                } as React.CSSProperties
```

Comment:
nit: I feel like this might be cleaner if we defined the styles as a var outside the render block